src/modules/mim/packagemanagerconfiguration.json

```diff
@@ -32,46 +25,18 @@
               },
               {
                 "name": "SourcesFingerprint",
+                "schema": "string"
+              },
+              {
+                "name": "SourcesFilenames",
                 "schema": {
-                  "type": "map",
-                  "mapKey":  {
-                    "name": "Id",
-                    "schema": "string"
-                  },
-                  "mapValue": {
-                    "name": "Fingerprint",
-                    "schema": "string"
-                  }
+                  "type": "array",
+                  "elementSchema": "string"
                 }
               },
               {
                 "name": "ExecutionState",
-                "schema": {
-                  "type": "enum",
-                  "valueSchema": "integer",
-                  "enumValues": [
-                    {
-                      "name": "Unknown",
-                      "enumValue": 0
-                    },
-                    {
-                      "name": "Running",
-                      "enumValue": 1
-                    },
-                    {
-                      "name": "Succeeded",
-                      "enumValue": 2
-                    },
-                    {
-                      "name": "Failed",
-                      "enumValue": 3
-                    },
-                    {
-                      "name": "TimedOut",
-                      "enumValue": 4
-                    }
-                  ]
-                }
+                "schema": "string"
```

Comment:
>"schema": "string" [](http://example.com/codeflow?start=16&length=18)

"State" implies a well defined enumeration of  values or similar. A string is not that. Why are you changing from the enum of integers to plain string? What will this string contain?

Reply:
The module's folder will contain a README.md explaining all the details. The PR is being worked on at the moment, if you want to catch an early [glimpse](https://github.com/Salazander/azure-osconfig/pull/16).

Generally speaking, our module has a general state and a substate. When an error occurs, we want to provide information about which package/source caused it. When running through the package installation process, we want to report back which package is being processed. This gives meaningful insights to the operator in case of healthy and unhealthy operating conditions. 

Packing this into one string allows a concise human- and machine-readable way to encode state and context.

Reply:
The MIM and DTDL have well defined structure to define such configuration. This helps the remote administrators. Otherwise all reported would be a single string.

Please split this string into meaningful MIM settings such as an integer for error|status code, enumeration of integers for reason, string for status description, etc.

Reply:
yes, we will make this change, updated MIM is coming soon.

Reply:
Looks good to me. Thanks